apps/passport-server/test/devconnect.spec.ts

```diff
@@ -0,0 +1,265 @@
+    );
+    // Should fail on duplicate (organizer_url, token)
+    try {
+      await sqlQuery(
+        application.context.dbPool,
+        "insert into pretix_organizers_config (organizer_url, token) values ('organizer-url-1', 'token2') returning id"
+      );
+      expect.fail();
+    } catch (e) {
+      // Should end up here
+    }
+    // Insert organizer 2
+    await sqlQuery(
+      application.context.dbPool,
+      "insert into pretix_organizers_config (organizer_url, token) values ('organizer-url-2', 'token2')"
+    );
+    expect(
+      (
+        await sqlQuery(
+          application.context.dbPool,
+          "select id from pretix_organizers_config"
+        )
+      ).rowCount
+    ).to.equal(2);
+  });
+
+  step("test events config", async function () {
+    // Insert organizer 1 and 2 from previous test
+    // Insert event 1
+    await sqlQuery(
+      application.context.dbPool,
+      "insert into pretix_events_config (pretix_organizers_config_id, event_id, active_item_ids) values (1, 'event-1', '{}')"
+    );
+    // Should fail on duplicate (event id, org id)
+    try {
+      await sqlQuery(
+        application.context.dbPool,
+        "insert into pretix_events_config (pretix_organizers_config_id, event_id, active_item_ids) values (1, 'event-1', '{}')"
+      );
+      expect.fail();
+    } catch (e) {
+      // Should end up here
+    }
+    // Insert it again with updated event-id
+    await sqlQuery(
+      application.context.dbPool,
+      "insert into pretix_events_config (pretix_organizers_config_id, event_id, active_item_ids) values (3, 'event-2', '{}')"
+    );
+
+    // Insert with active item IDs
+    await sqlQuery(
+      application.context.dbPool,
+      "insert into pretix_events_config (pretix_organizers_config_id, event_id, active_item_ids) values (1, 'event-3', '{123, 456}')"
+    );
+    expect(
+      (
+        await sqlQuery(
+          application.context.dbPool,
+          "select id from pretix_events_config"
+        )
+      ).rowCount
+    ).to.equal(3);
+  });
+});
+
+describe("devconnect functionality", function () {
+  this.timeout(15_000);
+
+  let application: PCDPass;
+  let _emailAPI: IEmailAPI;
+  let devconnectPretixMocker: DevconnectPretixDataMocker;
+  let _devconnectPretixSyncService: DevconnectPretixSyncService;
+  let db: Pool;
+
+  this.beforeAll(async () => {
+    await overrideEnvironment(pcdpassTestingEnv);
+    db = await getDB();
+
+    await sqlQuery(
+      db,
+      `
+    insert into pretix_organizers_config (organizer_url, token)
+    values ('organizer-url', 'token')
+    `
+    );
+    await sqlQuery(
+      db,
+      `
+    insert into pretix_events_config (pretix_organizers_config_id, event_id, active_item_ids)
+    values
+      (1, $1, '{ ${ITEM_1} }'),
+      (1, $2, '{ ${ITEM_1}, ${ITEM_2} }'),
+      (1, $3, '{}')
+    `,
+      [EVENT_A, EVENT_B, EVENT_C]
+    );
```

Comment:
i recommend initializing the state of the database configuration for events and items using explicit functions created for that purpose that we'll end up using in production

Reply:
added a TODO for this!